doc/nrf/drivers/nrf700x.rst

```diff
@@ -0,0 +1,69 @@
+.. _nrf700x_wifi:
+
+nRF700X Wi-Fi Driver
+####################
+
+.. contents::
+   :local:
+   :depth: 2
+
+This driver implements the Wi-Fi protocol for the nRF700X FullMAC series of devices. FullMAC devices implement the Wi-Fi
+protocol in the chipset, and the driver only needs to configure the chipset and transfer the frames to and from the device
+to networking stack.
+
+nRF700X is a companion chip and can be used with any SoC, primarily with Nordic Semiconductor's nRF53 and nRF91 Series SoCs.
+
+Driver can be enabled by using the :kconfig:option:`CONFIG_WIFI_NRF700X` Kconfig option.
+
+Block diagram
+*************
+
+Below is the block diagram of the nRF700X Wi-Fi driver.
+
+.. figure:: /images/nrf700x_wifi_driver.png
+   :alt: nRF700X Wi-Fi driver block diagram
+   :align: center
+   :figclass: align-center
+
+   nRF700X Wi-Fi driver block diagram
+
+Design overview
+***************
+The nRF700X Wi-Fi driver follows OS agnostic design, and the driver is implemented in to OS agnostic and OS (Zephyr) specific implementation files.
+The OS agnostic files are located in :file:`drivers/wifi/nrf700x/osal/` folder, and the OS specific files are located in :file:`drivers/wifi/nrf700x/zephyr/` folder.
+
+The driver is designed to be used with the Zephyr networking stack, and the driver is implemented as a network interface driver. Driver implements various IEEE082.11 Features
+and supports the following features:
+
+* Wi-Fi 6 (802.11ax) support
+* WPA3/WPA2 personal security
+* IEEE 802.11 power save modes
+* Scan only mode
+* IEEE 802.11 Station (STA) mode
+* IEEE 802.11 SoftAP mode - not supported yet
+* Wi-Fi Direct mode - not supported yet
+
+Except for scan  only mode, the driver uses `hostapd` to implement AP MLME and `wpa_supplicant` to implement 802.1X supplicant.
+
+Driver to nRF700X communication
+*******************************
+The driver communicates with the nRF700X device using the QSPI/SPI interface. The driver uses the QSPI/SPI interface to send commands to the nRF700X device, and to transfer the
+data to and from the device. nRF7002 DK uses QSPI where as nRF7002 EK uses SPI.
+
+For nRF7002 EK, a new shield `nrf7002_ek` is required to connect the nRF7002 EK to the SoC.
+
+
+Radio test mode
+***************
+The nRF700X Wi-Fi driver supports radio test mode, which can be used to test the RF performance of the nRF700X device. This is a build time option, and can be enabled by
+:kconfig:option:`CONFIG_NRF700X_RADIO_TEST` Kconfig option.
+
+The radio test mode is implemented as a separate application, and the application is located in :file:`samples/wifi/radio_test/` folder.
+
+API documentation
+*****************
+
+| Source folder: :file:`drivers/wifi/nrf700x/`
```

Comment:
Do we have any header file and source file to mention?

Reply:
there are too many, so, just pointed to the folder